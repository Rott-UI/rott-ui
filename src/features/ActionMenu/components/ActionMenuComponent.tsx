// React Imports
import React, {FC} from 'react'

// Component Imports
import {Item, Pressable, Label, Button, List, ActionMenu} from '../..'
import {ActionMenuStyles} from '../styles'
import {ActionMenuProps, ActionModel} from '../models'

// Constant Imports

// Util and Lib Imports

/**
 * Action Menu
 *
 * @param title Başlık alanı
 * @param subTitle Alt başlık alanı
 * @param data ActionModel[] formatında veri listesi
 * @param visible ActionMenu komponentin bulunduğu yerdeki Show state i
 * @returns Action Menu Renderlanır
 *
 */
export const ActionMenuComponent: FC<ActionMenuProps> = ({
  title,
  subTitle,
  data,
  itemHeight,
  maxItem,
  separatorTotalHeight,
}) => {
  const itemCount = (data?.length ?? 0) > maxItem! ? maxItem! : data?.length ?? 0

  // Listede renderlanacak Item
  const renderItem = ({
    title: renderItemTitle,
    name: renderItemName,
    action: onAction,
  }: ActionModel) => (
    <Pressable
      testID={`action-menu-data-${renderItemName}-test-id`}
      height={itemHeight}
      justifyContentCenter
      alignItemsCenter
      onPress={() => onAction()}>
      <Label fontSize='lg' variant='grey-900' fontWeight={500}>
        {renderItemTitle}
      </Label>
    </Pressable>
  )

  return (
    <Item size='full' style={ActionMenuStyles().actionMenuContainer} alignItemsCenter>
      <Item
        width={342}
        backgroundColor={COLOURS.GREY100}
        borderTopStartRadius={!(title || subTitle) ? 12 : 0}
        borderTopEndRadius={!(title || subTitle) ? 12 : 0}
        borderBottomStartRadius={12}
        borderBottomEndRadius={12}
        marginBottom={16}>
        <List
          testID='action-menu-test-id'
          style={{minHeight: 2}}
          width={342}
          height={itemHeight! * itemCount + separatorTotalHeight!}
          data={data}
          renderItem={({item}) => renderItem(item)}
          horizontal={false}
          separatorVariant='neutral-alpha-200'
          renderSeparator
          headerSeparator
          footerSeparator
          estimatedItemSize={display.normalize(itemHeight!, 'height')}
          showsVerticalScrollIndicator={false}
          scrollEnabled={(data?.length ?? 0) > maxItem!}
        />
      </Item>

      <Button
        variant='primary'
        onPress={() => ActionMenu.hideActionMenu()}
        fontSize='xl'
        size='full'
        testID='action-menu-cancel-test-id'
        marginBottom={21}>
        {translator('COMMON.CANCEL')}
      </Button>
    </Item>
  )
}
