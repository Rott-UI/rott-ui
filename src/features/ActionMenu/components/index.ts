export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
