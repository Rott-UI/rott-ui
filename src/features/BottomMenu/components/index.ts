export * from './BottomMenu'
export * from './BottomMenuItem'
