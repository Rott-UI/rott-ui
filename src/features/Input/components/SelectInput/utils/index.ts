export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
