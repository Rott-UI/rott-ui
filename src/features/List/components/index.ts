export * from './List'
export * from './FavoriteSwipeComponent'
