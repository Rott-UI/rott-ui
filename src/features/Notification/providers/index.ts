export * from './ToasterNotificationProvider'
export * from './NotificationProdiver'
