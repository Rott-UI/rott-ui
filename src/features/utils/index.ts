export * from './commonUiTestExtension'
export * from './commonUiStyleProperties'
