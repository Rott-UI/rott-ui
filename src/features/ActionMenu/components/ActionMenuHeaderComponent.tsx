// React Imports
import React, {FC} from 'react'

// Util and Lib Imports
import {Item, Label} from '../..'
import {themeConfig} from '../../../providers'

interface ActionMenuHeaderProps {
  title?: Nullable<string>
  subTitle?: Nullable<string>
}

export const ActionMenuHeaderComponent: FC<ActionMenuHeaderProps> = ({title, subTitle}) => {
  return (
    <Item size='full' alignItemsCenter>
      <Item testID='action-menu-header-test-id' width={342}>
        {(title || subTitle) && (
          <Item
            justifyContentCenter
            alignItemsCenter
            size='full'
            backgroundColor={themeConfig.colors['grey-100']}
            borderTopStartRadius={12}
            borderTopEndRadius={12}
            height={54}>
            {title && (
              <Label
                fontSize='xl'
                variant='grey-900'
                fontWeight={700}
                marginBottom={subTitle ? 4 : undefined}>
                {title}
              </Label>
            )}
            {subTitle && (
              <Label fontSize='sm' variant='grey-900' fontWeight={500}>
                {subTitle}
              </Label>
            )}
          </Item>
        )}
      </Item>
    </Item>
  )
}
