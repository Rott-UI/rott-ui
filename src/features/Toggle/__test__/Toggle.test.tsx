// React Imports
import React from 'react'

// Component Imports
import {Toggle} from '../components'

// Util and Lib Imports
import {act, fireEvent, render, waitFor} from './../../../utils'

// Provider Imports
import {themeConfig} from '../../../providers'

describe('Toggle -> Custom Component', () => {
  const toggleContainerTestId = 'toggle-container-test-id'
  const toggleTestId = 'toggle-test-id'

  it('toggle ilk render anında snapshot ile eşleşmeli', () => {
    const renderedToggle = render(<Toggle isOn={false} />)

    expect(renderedToggle).toMatchSnapshot()
  })

  it('toggle default propslar ile render olmalı', async () => {
    const {getByTestId} = render(<Toggle isOn={false} />)

    await act(() => {
      const toggleContainer = getByTestId(toggleContainerTestId)

      expect(toggleContainer).toBeDefined()
    })
    await act(() => {
      const toggleWheel = getByTestId(toggleTestId)

      expect(toggleWheel).toBeDefined()
    })
  })

  it('tıklandığında onToggleChange çağırılmalı', async () => {
    const onToggleMock = jest.fn()
    const {getByTestId} = render(<Toggle onToggleChange={onToggleMock} isOn={false} />)

    await act(() => {
      const toggleContainer = getByTestId(toggleContainerTestId)

      fireEvent.press(toggleContainer)
    })

    await waitFor(() => {
      expect(onToggleMock).toHaveBeenCalled()
    })
  })

  describe('isOn özelliği değiştiğinde stil özellikleri değişmeli', () => {
    const onToggleMock = jest.fn()

    it('pasif ise', async () => {
      const {getByTestId} = render(<Toggle onToggleChange={onToggleMock} isOn={false} />)

      await act(() => {
        const toggleContainer = getByTestId(toggleTestId)

        expect(toggleContainer).toHaveStyle({
          backgroundColor: themeConfig?.colors?.['grey-200'],
        })
      })
    })
    it('aktif ise', async () => {
      const {getByTestId} = render(<Toggle onToggleChange={onToggleMock} isOn={true} />)

      await act(() => {
        const toggleContainer = getByTestId(toggleTestId)

        expect(toggleContainer).toHaveStyle({
          backgroundColor: themeConfig?.colors?.primary,
        })
      })
    })
  })
})
