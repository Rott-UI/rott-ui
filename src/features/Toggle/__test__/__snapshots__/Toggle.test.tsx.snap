// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle -> Custom Component toggle ilk render anında snapshot ile eşleşmeli 1`] = `
<View
  alignItemsCenter={true}
  height={32}
  maxHeight={32}
  maxWidth={64}
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "bottom": undefined,
      "flex": undefined,
      "flexDirection": "row",
      "flexGrow": undefined,
      "gap": undefined,
      "height": 62,
      "justifyContent": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": 62,
      "maxWidth": 123,
      "minHeight": undefined,
      "minWidth": undefined,
      "overflow": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
      "position": undefined,
      "right": undefined,
      "top": undefined,
      "width": 123,
      "zIndex": undefined,
    }
  }
  width={64}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animated={false}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "bottom": undefined,
        "flex": 1,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      }
    }
    testID="toggle-container-test-id"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(161, 173, 175, 1)",
          "borderRadius": 16,
          "height": 62,
          "justifyContent": "center",
          "width": 123,
        }
      }
      testID="toggle-test-id"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderColor": "rgba(34, 63, 70, 0.15)",
            "borderRadius": 24,
            "borderWidth": 4,
            "height": 44,
            "marginLeft": 8,
            "width": 44,
          }
        }
      />
    </View>
  </View>
</View>
`;
